packages/neuron-wallet/src/block-sync-renderer/sync/light-connector.ts

```diff
@@ -181,27 +181,35 @@ export default class LightConnector extends Connector<CKBComponents.Hash> {
         }))
       })
       .flat()
-      .concat(appendScripts ?? [])
     const walletMinBlockNumber = await SyncProgressService.getWalletMinBlockNumber()
     const wallets = await WalletService.getInstance().getAll()
     const walletStartBlockMap = wallets.reduce<Record<string, string | undefined>>(
       (pre, cur) => ({ ...pre, [cur.id]: cur.startBlockNumberInLight }),
       {}
     )
-    const setScriptsParams = allScripts.map(v => ({
-      ...v,
-      blockNumber:
-        existSyncscripts[scriptToHash(v.script)]?.blockNumber ??
-        walletStartBlockMap[v.walletId] ??
-        `0x${(walletMinBlockNumber?.[v.walletId] ?? 0).toString(16)}`
-    }))
+    const otherTypeSyncProgress = await SyncProgressService.getOtherTypeSyncProgress()
+    const setScriptsParams = [
+      ...allScripts.map(v => ({
+        ...v,
+        blockNumber:
+          existSyncscripts[scriptToHash(v.script)]?.blockNumber ??
+          walletStartBlockMap[v.walletId] ??
+          `0x${(walletMinBlockNumber?.[v.walletId] ?? 0).toString(16)}`
+      })),
+      ...appendScripts.map(v => ({
+        ...v,
+        blockNumber:
+          existSyncscripts[scriptToHash(v.script)]?.blockNumber ??
+          `0x${(otherTypeSyncProgress[scriptToHash(v.script)] ?? 0).toString(16)}`
+      }))
+    ]
     await this.lightRpc.setScripts(setScriptsParams)
-    await SyncProgressService.resetSyncProgress(allScripts)
     const walletIds = [...new Set(this.addressMetas.map(v => v.walletId))]
-    await SyncProgressService.removeWalletsByExists(walletIds)
+    await SyncProgressService.resetSyncProgress([...allScripts, ...appendScripts])
```

Comment:
Combination of arrays with spreading is used here and somewhere, the readability could be improved by `[allScripts, appendScripts].flat()`

Just a suggestion for code style